### PR TITLE
Add a section for order request culture providers

### LINKED ISSUE #21614: Further explanation not provided
Further explanation is not provided.

> Later in the article we'll see how you can change the order and even add a custom culture provider

There was no further explanation later in the article on how to change the order of the culture providers.

[Related issue](https://github.com/dotnet/AspNetCore.Docs/issues/15532)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d1833676-ff1b-c587-cc9c-81afffa8ef68
* Version Independent ID: ad989395-3f49-4d55-521e-cf6fe56281ee
* Content: [Globalization and localization in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/localization?view=aspnetcore-5.0)
* Content Source: [aspnetcore/fundamentals/localization.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/localization.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #21614

### CODE REVIEW

**Comment by hishamco** [comment]
@Rick-Anderson please wait, I notice something strange in the docs, so I need to verify before the merge

**Comment by hishamco** [comment]
In `Use a custom provider` section there was an example I wrote before ASP.NET Core introduced `AddInitialRequestCultureProvider`, I'm not sure why I didn't find this anymore. Any idea?

/cc @guardrex @guardrex

File: aspnetcore/fundamentals/localization.md [hunk 1]
```diff
@@ -359,6 +359,23 @@ services.Configure<RequestLocalizationOptions>(options =>
 
 Use `RequestLocalizationOptions` to add or remove localization providers.
 
+### Change request culture providers order
+
+As mentioned earlier, `RequestLocalizationOptions` came up with three default request cultures provider: `QueryStringRequestCultureProvider`, `CookieRequestCultureProvider` and `AcceptLanguageHeaderRequestCultureProvider` respectively. In case you need to change the order of these providers you could use `RequestLocalizationOptions.RequestCultureProviders` property as shown in the example below:
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/localization.md:364]:
```suggestion
<xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptions> has three default request culture providers: <xref:Microsoft.AspNetCore.Localization.QueryStringRequestCultureProvider>, <xref:Microsoft.AspNetCore.Localization.CookieRequestCultureProvider>, and <xref:Microsoft.AspNetCore.Localization.AcceptLanguageHeaderRequestCultureProvider>. Use [`RequestLocalizationOptions.RequestCultureProviders`]](xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptions.RequestCultureProviders) property to change the order of these providers as shown in the following below:
```

File: aspnetcore/fundamentals/localization.md [hunk 2]
```diff
@@ -359,6 +359,23 @@ services.Configure<RequestLocalizationOptions>(options =>
...
+```csharp
+    app.UseRequestLocalization(options =>
+    {
+        var questStringCultureProvider = options.RequestCultureProviders[0];    
+        options.RequestCultureProviders.RemoveAt(0);
+        options.RequestCultureProviders.Insert(1, questStringCultureProvider);
+    });
+```
+
+In the example above, we switch between the `QueryStringRequestCultureProvider` and `CookieRequestCultureProvider`, so the `RequestLocalizationMiddleware` will look for the cultures from the cookies first, then query string. Perhaps the example is silly and simple, but image that you can order these providers with your custom once according your needs.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/localization.md:375]:
```suggestion
In the preceding example above, the  order of `QueryStringRequestCultureProvider` and `CookieRequestCultureProvider` is swithched, so the `RequestLocalizationMiddleware` looks for the cultures from the cookies first, then query string.
```

File: aspnetcore/fundamentals/localization.md [hunk 3]
```diff
@@ -359,6 +359,23 @@ services.Configure<RequestLocalizationOptions>(options =>
...
+
+Another real use case - that we mentioned above - is when we add a custom provider via `AddInitialRequestCultureProvider` it sets the order to `0`, so this provider will take the precedence over the others.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/localization.md:377]:
```suggestion
As previously mentioned, add a custom provider via <xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptionsExtensions.AddInitialRequestCultureProvider%2A> which sets the order to `0`, so this provider takes the precedence over the others.
```

**Review by Rick-Anderson — APPROVED** [review]

**Review by Rick-Anderson — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
@hishamco  fantastic work, much appreciated.

**Comment by hishamco** [comment]
@Rick-Anderson what about my comment [above](https://github.com/dotnet/AspNetCore.Docs/pull/24404#issuecomment-1000492915)

**Comment by Rick-Anderson** [comment]
@hishamco  do you want to create another issues to document `AddInitialRequestCultureProvider` and a PR to close it?

**Comment by hishamco** [comment]
`AddInitialRequestCultureProvider` is already there, I mean how to that before we add this API, It was there, but I don't know why it's deleted. I need to revise the history and check what's happen

Thanks

**Comment by hishamco** [comment]
@Rick-Anderson seems the snippets / examples before 3.0 has been removed, right?

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson seems the snippets / examples before 3.0 has been removed, right?

Yes, only 3.1 and higher is supported.

**Comment by hishamco** [comment]
I see that's why I surprising when  saw sample that I did long time back, thanks for clarification

**Pull request merged**